pkg/router/route_group.go

```diff
@@ -46,64 +81,134 @@ type RouteGroup struct {
 	done    chan struct{}
 	once    sync.Once
 
-	rt routing.Table
-
-	lastSent int64
+	readTimer  *time.Timer
+	writeTimer *time.Timer
+	// TODO: try to implement timed out flags with open/closed chan struct{}
+	//  in order to be able to check timeout in select statement
+	readTimedOut  atomicbool.Bool // set true when read deadline has been reached
+	writeTimedOut atomicbool.Bool // set true when write deadline has been reached
 }
 
-func NewRouteGroup(rt routing.Table, desc routing.RouteDescriptor) *RouteGroup {
+func NewRouteGroup(cfg *RouteGroupConfig, rt routing.Table, desc routing.RouteDescriptor) *RouteGroup {
+	if cfg == nil {
+		cfg = DefaultRouteGroupConfig()
+	}
+
 	rg := &RouteGroup{
+		logger:  logging.MustGetLogger(fmt.Sprintf("RouteGroup %s", desc.String())),
 		desc:    desc,
+		rt:      rt,
+		tps:     make([]*transport.ManagedTransport, 0),
 		fwd:     make([]routing.Rule, 0),
 		rvs:     make([]routing.Rule, 0),
-		tps:     make([]*transport.ManagedTransport, 0),
-		readCh:  make(chan []byte, readChBufSize),
+		readCh:  make(chan []byte, cfg.ReadChBufSize),
 		readBuf: bytes.Buffer{},
-		rt:      rt,
+		done:    make(chan struct{}),
 	}
 
-	go rg.keepAliveLoop()
+	go rg.keepAliveLoop(cfg.KeepAliveInterval)
 
 	return rg
 }
 
 // Read reads the next packet payload of a RouteGroup.
-// The Router, via transport.Manager, is responsible for reading incoming packets and pushing it to the appropriate RouteGroup via (*RouteGroup).readCh.
-// To help with implementing the read logic, within the dmsg repo, we have ioutil.BufRead, just in case the read buffer is short.
+// The Router, via transport.Manager, is responsible for reading incoming packets and pushing it
+// to the appropriate RouteGroup via (*RouteGroup).readCh.
+// To help with implementing the read logic, within the dmsg repo, we have ioutil.BufRead,
+// just in case the read buffer is short.
+// TODO: too long, simplify
```

Comment:
Yes, the logic here needs simplifications, and we need to use the `bytes.Buffer` just in case we have short reads.

Reply:
Commented it out to ensure the code without it works as expected. Now it does, `bytes.Buffer` is back.